recipe/conda_build_config.yaml

```diff
@@ -1,2 +1,4 @@
+c_stdlib_version:          # [osx and x86_64]
+  - 10.11                  # [osx and x86_64]
 MACOSX_DEPLOYMENT_TARGET:  # [osx and x86_64]
   - 10.11                  # [osx and x86_64]
```

Comment:
AIUI macOS 10.11 was added as part of PR ( https://github.com/conda-forge/yt-feedstock/pull/69 ) to build with a newer macOS version than conda-forge used at the time (previously 10.9). Since then conda-forge has bumped the minimum macOS version to 10.13 ( https://github.com/conda-forge/conda-forge.github.io/issues/1844 ), which means the 10.11 version used here is now old in comparison

Have made the necessary changes to keep macOS 10.11 working here ( particularly with the addition of `{{ stdlib("c") }}`: https://github.com/conda-forge/conda-forge.github.io/issues/2102 ). That said, maybe it is worth revisiting whether this 10.11 pin is still needed or if 10.13 would suffice, in which case this `conda_build_config.yaml` file could be dropped entirely. This need not happen in this PR. Just mentioning it as the fixes needed to keep 10.11 working here are happening in this PR